src/AstGen.zig

```diff
@@ -6097,7 +6098,21 @@ fn identifier(
             },
             .gen_zir => s = s.cast(GenZir).?.parent,
             .defer_normal, .defer_error => s = s.cast(Scope.Defer).?.parent,
-            .namespace, .top => break, // TODO look for ambiguous references to decls
+            // look for ambiguous references to decls
+            .namespace => {
```

Comment:
Can you add a test case that ensures that this logic has the correct behavior, in the case that this loop escapes out of the namespace, and then there is a local_val or local_ptr with a matching name?

Something like this:

```zig
test {
    var bar: i32 = 1234;
    const S = struct {
        fn baz() i32 {
            return bar;
        }
    };
    var quux = S.baz();
}
```